File: 3577-count-the-number-of-computer-unlocking-permutations/3577-count-the-number-of-computer-unlocking-permutations.cpp
#define MOD 1000000007
class Solution {
public:
    int countPermutations(vector<int>& complexity) {
        for(int i = 1; i < complexity.size(); i++)
        {
            if(complexity[i] <= complexity[0])
            {
                return 0;
            }
        }
        return fact(complexity.size() - 1);
    }
private:
    int fact(int n)
    {
        if(n == 0 || n == 1)
        {
            return 1;
        }
        return (n * 1LL * fact(n - 1)) % MOD;
    }
};

File: 3577-count-the-number-of-computer-unlocking-permutations/3577-count-the-number-of-computer-unlocking-permutations.py
class Solution:
    def fact(self, n: int) -> int:
        if n == 0 or n == 1:
            return 1
        return (n * self.fact(n - 1)) % 1000000007
    def countPermutations(self, complexity: List[int]) -> int:
        for c in complexity[1:]:
            if c <= complexity[0]:
                return 0
        return self.fact(len(complexity) - 1)
        